compiler/semcall.nim

```diff
@@ -265,9 +263,9 @@ proc presentFailedCandidates(c: PContext, n: PNode, errors: CandidateErrors):
 const
   errTypeMismatch = "type mismatch: got <"
   errButExpected = "but expected one of: "
-  errUndeclaredField = "undeclared field: '$1'"
-  errUndeclaredRoutine = "attempting to call undeclared routine: '$1'"
-  errBadRoutine = "attempting to call routine: '$1'$2"
+  errUndeclaredField = "undeclared field: $1"
```

Comment:
I tried your PR again, it's starting to look good!
- [ ] bad:
```nim
when true:
  var a: string
  a = 1+2
```
XDG_CONFIG_HOME= $nimb r -d:case2 main # ok
XDG_CONFIG_HOME= $nimb r -d:case2  --hint:Source main # bad

```
t11525.nim(12, 8) Error: type mismatch: got int literal(3) for 3 but expected string
    a = 1+2
         ^
```
`a = 1+2` isn't colorized

Reply:
I didnt even realize that `--hint:source` was an option.

Reply:
Try this with some Regex and PEG and see if it crashes.